src/Provider/AbstractProvider.php

```diff
@@ -707,7 +707,7 @@ protected function parseResponse(ResponseInterface $response)
      */
     protected function prepareAccessTokenResponse(array $result)
     {
-        if ($this->getAccessTokenResourceOwnerId()) {
+        if ($this->getAccessTokenResourceOwnerId() !== null) {
```

Comment:
This doesn't seem right, because `false` and `''` would not be valid possibilities.